src/ObjectStateMutations.ts

```diff
@@ -177,6 +177,12 @@ function nestedSet(obj, key, value) {
   }
 }
 
+function isJSONArray(val) {
```

Comment:
Nice! Very clean. Can you add a comment on what a JSON Array is. You can move the one in `commitServerChange`